File: Unidad1/diptongos.py

"""
Programa Diptongos
Autor: Héctor Rodríguez

Este programa permite obtener los
diptongos existentes en un archivo
de texto. Muestra en pantalla el diptongo,
la palabra donde se encuentra y en que
linea del archivo.

Esta version en Python reconocé vocales con
tilde, y diéresis.
"""

vocales = ['a', 'e', 'i', 'o', 'u', 'á', 'é', 'í', 'ó', 'ú', 'ö', 'ü']

try:
    with open('parrafo.txt', encoding="utf-8") as file:
        lineNumber = 1
        for line in file:
            words = line.split(" ")
            for word in words:
                vocalAntes = ''
                for char in word:
                    if char in vocales:
                        if not vocalAntes == '' and not vocalAntes == char:
                            print('\nPalabra:', word)
                            print('Diptongo: {}{}'.format(vocalAntes, char))
                            print('En la linea', lineNumber)
                        vocalAntes = char
                    else:
                        vocalAntes = ''
            lineNumber += 1
except Exception as e:
    print("Error al abrir el archivo")

File: Unidad1/Diptongos.java

import java.io.File; 
import java.io.FileNotFoundException;
import java.util.Scanner; 
import java.util.logging.Level;
import java.util.logging.Logger;

/*

Programa Diptongos
Autor: Héctor Rodríguez

Este programa permite obtener los
diptongos existentes en un archivo
de texto. Muestra en pantalla el diptongo,
la palabra donde se encuentra y en que
linea del archivo.

Esta version en Java sólo reconoce
vocales sin tilde o diéresis

 */
public class Diptongos {

    public static void main(String[] args) {   
        String vocales = "aeiou";
        
        File file = new File("parrafo.txt"); 
        Scanner sc = null; 
        
        try {
            sc = new Scanner(file);
        } catch (FileNotFoundException ex) {
            System.out.println("No se pudo encontrar el archivo");
        } catch (Exception e) {
            System.out.println("Error grave al importar el archivo");
        }

        int lineNumber = 1;
        
        while (sc.hasNextLine()) {
            String[] palabras = sc.nextLine().split(" ");
            for(String palabra : palabras){
                char vocalAntes = ' ';
                for(char letra : palabra.toCharArray()) {
                    if (vocales.contains(String.valueOf(letra))) {
                        if ((vocalAntes != ' ') && (vocalAntes != letra)) {
                            System.out.println("\nPalabra: " + palabra);
                            System.out.println("Diptongo: " + vocalAntes + letra);
                            System.out.println("En la linea: " + lineNumber);
                        }
                        vocalAntes = letra;
                    } else {
                        vocalAntes = ' ';
                    }
                }
            }
            lineNumber++;
        }
    }
    
}